python/ray/tune/tests/test_searchers.py

```diff
@@ -0,0 +1,155 @@
+import unittest
+
+import numpy as np
+
+from ray import tune
+
+
+def _invalid_objective(config):
+    # DragonFly uses `point`
+    metric = "point" if "point" in config else "report"
+
+    if config[metric] > 4:
+        tune.report(float("inf"))
+    elif config[metric] > 3:
+        tune.report(float("-inf"))
+    elif config[metric] > 2:
+        tune.report(np.nan)
+    else:
+        tune.report(float(config[metric]) or 0.1)
+
+
+class InvalidValuesTest(unittest.TestCase):
+    def setUp(self):
+        self.config = {"report": tune.uniform(0.0, 5.0)}
+
+    def tearDown(self):
+        pass
+
```

Comment:
to speed this up, can you do:

```
    @classmethod
    def setUpClass(cls):
        ray.init(
            num_cpus=4, num_gpus=0, local_mode=True, include_dashboard=False)

    @classmethod
    def tearDownClass(cls):
        ray.shutdown()
        _register_all()
```

maybe you'll need to remove the `reuse_actors`.